python/nvfuser_direct/__init__.py

```diff
@@ -42,25 +43,35 @@ def __init__(self):
         # Monkey patching nvfuser_direct.ops submodule to mimic python_frontend
         # FusionDefinition.ops API. This is to maintain backwards compatibilty.
         self.ops = _C_DIRECT.ops
-        self.fusion = _C_DIRECT.Fusion()
-        self.fusion_guard = None
+        self._fusion = None
+        self._fusion_guard = None
+
+    @property
+    def fusion(self):
```

Comment:
Created property for `Fusion`. Before compilation, refer to python `_fusion`. After compilation, refer to cpp `FusionExecutorCache` fusion